### DISCUSSION TITLE
ID3 tags for HLS Playback

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#5795] [General]
Hello,
It seems that ID3 tags are supported for HLS playback now according to the release logs [release logs](https://github.com/ant-media/Ant-Media-Server/releases) for Ant Media Server version 2.7.0

How can I make use of ID3 tags for the HLS playback and are there any additional set of steps/configurations that I need to do in order to use this feature?
Also, is this available for both the Community edition as well as the Enterprise Edition as both support HLS playback?

Thank you

**Comment by burak-58** [comment]
With ID3 feature you can send data to HLS viewers. For now it sends the messages which is provided by REST method. You use this feature you should enable ID3 feature in application settings. [Here](https://antmedia.io/javadoc/io/antmedia/AppSettings.html#id3TagEnabled
) is the related setting.
To add an ID3 text data in to a stream, you can call [this REST method](https://antmedia.io/rest/#/BroadcastRestService/addID3Data
)
Here is a sample usage:
`curl -i -X POST -H "Accept: Application/json" -H "Content-Type: application/json" "https://{AMS-URL}:5443/{APP_NAME}/rest/v2/broadcasts/{STREAM_ID}/id3" -d '{TEXT}'
`

[Here ](https://codepen.io/Burak-Kekec/pen/PoXYMyG ) is a sample video player which receives the data and print it over the video.

**Reply by tepaze** [reply]
Thank you for your sample usage / player, and thanks for adding this feature.

I tested it with your sample video player. It's work fine, but when i make a call every 100ms, i just receive 1 data on each chunck of HLS...

This maybe normal, but in my case I'd like to call and receive data at least every 0.5s. Is there any way to do that ?

**Reply by Mohit-3196** [reply]
Thank you @tepaze for your comment.
Let me invite the developer @burak-58 to answer this for you.

Thank you

**Reply by burak-58** [reply]
Hi @tepaze,
Thanks for your interest. I will check and write you soon. BTW can you tell us about your use case?

**Reply by tepaze** [reply]
Hi @burak-58 
I'd like to inject data in a 100ms step to synchronise some component between the video and the player. It can be some smiley or  comment for example, but even many more things. My goal is to be as close as possible to the moment of the video.

**Reply by tepaze** [reply]
Hi @burak-58 
This is a bit old... Maybe you forgot my question :-) I would like to ingest metadata every 100-500ms, and actually I only get 1 metadata for each HLS chunk. .. So if I have a 2sec chunk, I get 1 metadata every 2sec...
Can you tell me if I'm making a mistake?
Best

**Reply by burak-58** [reply]
I will work on ID3 implementation more on this week. Let me write an update next week.
